app/controllers/api/v1/conversations_controller.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class ConversationsController < Api::V1::BaseController
+      include ConversationsHelper
+
+      before_action only: %i[create index show] do
+        require_access_token %w[read]
+      end
+
+      before_action only: %i[create destroy] do
+        require_access_token %w[read write]
+      end
+
+      rescue_from ActiveRecord::RecordNotFound do
+        render json: I18n.t("api.endpoint_errors.conversations.not_found"), status: :not_found
+      end
+
+      def index
+        params.permit(:only_after, :only_unread)
+        mapped_params = {}
+        mapped_params[:only_after] = params[:only_after] if params.has_key?(:only_after)
+        mapped_params[:unread] = params[:only_unread] if params.has_key?(:only_unread)
+        conversations_query = conversation_service.all_for_user(mapped_params)
+        conversations_page = pager(conversations_query, "conversations.created_at").response
+        conversations_page[:data] = conversations_page[:data].map {|x| conversation_as_json(x) }
+        render json: conversations_page
+      end
+
+      def show
+        conversation = conversation_service.find!(params[:id])
+        render json: conversation_as_json(conversation)
+      end
+
+      def create
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for create is too high. [26.15/20]